inSociety/Auth & Sign in/LoginViewController.swift

```diff
@@ -0,0 +1,184 @@
+                                titleColor: .mainYellow(), backgroundColor: nil)
+    
+    
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setUpViews()
+        
+        loginButton.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        signUpButton.addTarget(self, action: #selector(signUpButtonPressed), for: .touchUpInside)
+        googleButton.addTarget(self, action: #selector(googleButtonPressed), for: .touchUpInside)
+        
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        scrollView.contentSize = view.frame.size
+    }
+    
+}
+
+
+
+//MARK: - Buttons realization
+extension LoginViewController {
+    @objc private func loginButtonPressed() {
+        AuthService.shared.login(email: emailTextField.text,
+                                 password: passwordTextField.text) { result in
+            switch result {
+            case .success(let user):
+                FirestoreService.shared.getUserData(user: user) { result in
+                    switch result {
+                    case .success(let currentUser):
+                        let mainTabBar = MainTabBarController(currentUser: currentUser)
+                        mainTabBar.modalPresentationStyle = .fullScreen
+                        self.present(mainTabBar, animated: true)
+                    case .failure(_):
+                        self.present(SetupProfileViewController(currentUser: user), animated: true)
+                    }
+                }
+            case .failure(let error):
+                self.showAlert(with: "Failure", and: error.localizedDescription)
+            }
+        }
+    }
+    
+    @objc private func signUpButtonPressed() {
+        dismiss(animated: true) {
+            self.delegate?.toSignUpVC()
+        }
+    }
+    
+    @objc private func googleButtonPressed() {
+        let clientID = FirebaseApp.app()?.options.clientID
+        AuthService.shared.googleLogin(clientID: clientID, presentingVC: self) { result in
+            switch result {
+            case .success(let user):
+                FirestoreService.shared.getUserData(user: user) { result in
+                    switch result {
+                    case .success(let userModel):
+                        let main = MainTabBarController(currentUser: userModel)
+                        main.modalPresentationStyle = .fullScreen
+                        self.present(main, animated: true)
+                    case .failure(_):
+                        self.showAlert(with: "You have successfully registrated", and: nil) {
+                            self.present(SetupProfileViewController(currentUser: user), animated: true)
+                        }
+                    }
+                }
+            case .failure(let failure):
+                self.showAlert(with: "Error", and: failure.localizedDescription)
+            }
```

Comment:
this part is the same as in "loginButtonPressed()" so you can move it to different method and do not have 2 same code parts